ScanCommand/src/main/java/io/ballerina/scan/ScanCmd.java

```diff
@@ -0,0 +1,133 @@
+/*
+ *
+ *  * Copyright (c) 2024, WSO2 LLC. (https://www.wso2.com).
+ *  *
+ *  * WSO2 LLC. licenses this file to you under the Apache License,
+ *  * Version 2.0 (the "License"); you may not use this file except
+ *  * in compliance with the License.
+ *  * You may obtain a copy of the License at
+ *  *
+ *  *   http://www.apache.org/licenses/LICENSE-2.0
+ *  *
+ *  * Unless required by applicable law or agreed to in writing,
+ *  * software distributed under the License is distributed on an
+ *  * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  * KIND, either express or implied. See the License for the
+ *  * specific language governing permissions and limitations
+ *  * under the License.
+ *
+ */
+
+package io.ballerina.scan;
+
+import io.ballerina.cli.BLauncherCmd;
+import io.ballerina.scan.utilities.StringToListConverter;
+import picocli.CommandLine;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.PrintStream;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.List;
+
+@CommandLine.Command(name = "scan", description = "Perform static code analysis for ballerina packages")
+public class ScanCmd implements BLauncherCmd {
+
+    private final PrintStream outputStream;
+    @CommandLine.Parameters(description = "Program arguments")
```

Comment:
What is this for?

Reply:
The arglist attribute is used to retrieve the Ballerina file name (e.g., main.bal) when the scan command is executed like `bal scan main.bal` for single file projects